x-pack/filebeat/input/awss3/s3.go

```diff
@@ -93,7 +93,7 @@ func newS3Poller(log *logp.Logger,
 }
 
 func (p *s3Poller) handlePurgingLock(info s3ObjectInfo, isStored bool) {
-	id := info.name + info.key
+	id := info.name + info.key + info.etag + info.lastModified.String()
```

Comment:
Can we create a `stateID(bucket, key, etag, lastModified)` function to return the state ID. It's important that this ID is formed the same way everywhere so having a helper function will ensure consistency. The same code is also at used at https://github.com/elastic/beats/blob/9c513d85d9e7d497b5de9a1e93a66458b883def0/x-pack/filebeat/input/awss3/state.go#L44.